examples/mamba.py

```diff
@@ -0,0 +1,472 @@
+import os, sys, math, argparse
+
+sys.path.append(os.getcwd())
+from typing import Any, Optional
+from dataclasses import dataclass, field
+
+from tinygrad import Tensor, dtypes, nn
+from tinygrad.jit import TinyJit
+from tinygrad.helpers import fetch
+from tinygrad.nn.state import get_state_dict, load_state_dict, torch_load
+
+from extra.models.llama import RMSNorm
+from transformers import AutoTokenizer
+from einops import rearrange, repeat
+import numpy as np
+
+MODELS = {
+  "130m": {
+    "dim": 768,
+    "n_layers": 24,
+    "vocab_size": 50277,
+    "pad_vocab_size_multiple": 8,
+  },
+  "370m": {
+    "dim": 1024,
+    "n_layers": 48,
+    "vocab_size": 50277,
+    "pad_vocab_size_multiple": 8,
+  },
+  "790m": {
+    "dim": 1536,
+    "n_layers": 48,
+    "vocab_size": 50277,
+    "pad_vocab_size_multiple": 8,
+  },
+  "1.4b": {
+    "dim": 2048,
+    "n_layer": 48,
+    "vocab_size": 50277,
+    "pad_vocab_size_multiple": 8,
+  },
+  "2.8b": {
+    "dim": 2560,
+    "n_layer": 64,
+    "vocab_size": 50277,
+    "pad_vocab_size_multiple": 8,
+  },
+}
+
+
+def fetch_weights(model_name: str):
+  if model_name not in MODELS.keys():
+    raise Exception(f"Requested unknown mamba model: {model_name}")
+  downloaded = fetch(
+    f"https://huggingface.co/state-spaces/mamba-{model_name}/resolve/main/pytorch_model.bin?download=true"
+  )
+  weights = torch_load(downloaded)
+  return weights
+
+
+def selective_scan_ref(
```

Comment:
this can be renamed to `selective_scan`. the `_ref ` in the original code means "reference implementation" and is used to differ from the CUDA implementations